tests/0056-balanced_group_mt.c

```diff
@@ -0,0 +1,297 @@
+                                                            part.partition);
+                        rd_kafka_queue_forward(rkqu, NULL);
+                        tids[part.partition] = spawn_thread(rkqu);
+                }
+
+                rebalanced = 1;
+
+                break;
+
+        case RD_KAFKA_RESP_ERR__REVOKE_PARTITIONS:
+                if (assign_cnt == 0)
+                        TEST_FAIL("asymetric rebalance_cb");
+                assign_cnt--;
+                rd_kafka_assign(rk, NULL);
+                mtx_lock(&lock);
+                consumers_running = 0;
+                mtx_unlock(&lock);
+
+                break;
+
+        default:
+                TEST_FAIL("rebalance failed: %s", rd_kafka_err2str(err));
+                break;
+        }
+}
+
+static void get_assignment (rd_kafka_t *rk_c) {
+        while (!rebalanced) {
+                rd_kafka_message_t *rkmsg;
+                rkmsg = rd_kafka_consumer_poll(rk_c, 500);
+                if (rkmsg)
+                        rd_kafka_message_destroy(rkmsg);
+        }
+}
+
+int main_0056_balanced_group_mt (int argc, char **argv) {
+        const char *topic = test_mk_topic_name(__FUNCTION__, 1);
+        rd_kafka_t *rk_p, *rk_c;
+        rd_kafka_topic_t *rkt_p;
+        int msg_cnt = 1000;
+        int msg_base = 0;
+        int partition_cnt = 2;
+        int partition;
+        uint64_t testid;
+        rd_kafka_topic_conf_t *default_topic_conf;
+        rd_kafka_topic_partition_list_t *sub, *topics;
+        rd_kafka_resp_err_t err;
+        test_timing_t t_assign, t_close, t_consume;
+        int i;
+
+        exp_msg_cnt = msg_cnt * partition_cnt;
+
+        testid = test_id_generate();
+
+        /* Produce messages */
+        rk_p = test_create_producer();
+        rkt_p = test_create_producer_topic(rk_p, topic, NULL);
+
+        for (partition = 0; partition < partition_cnt; partition++) {
+                test_produce_msgs(rk_p, rkt_p, testid, partition,
+                                  msg_base + (partition * msg_cnt), msg_cnt,
+                                  NULL, 0);
+        }
+
+        rd_kafka_topic_destroy(rkt_p);
+        rd_kafka_destroy(rk_p);
+
+        if (mtx_init(&lock, mtx_plain) != thrd_success)
+                TEST_FAIL("Cannot create mutex.");
+
+        test_conf_init(NULL, &default_topic_conf,
+                       (test_session_timeout_ms * 3) / 1000);
+
+        test_topic_conf_set(default_topic_conf, "auto.offset.reset",
+                            "smallest");
+
+        /* Fill in topic subscription set */
+        topics = rd_kafka_topic_partition_list_new(1);
+        rd_kafka_topic_partition_list_add(topics, topic, RD_KAFKA_PARTITION_UA);
+
+        /* Create consumers and start subscription */
+        rk_c = test_create_consumer(
+                topic /*group_id*/, rebalance_cb, NULL,
+                rd_kafka_topic_conf_dup(default_topic_conf));
```

Comment:
You dont need to dup this since you are destroying it just below.
Loose the dup and the destroy